Public/Src/Cache/ContentStore/Distributed/Utilities/GrpcDistributedPathTransformer.cs

```diff
@@ -42,9 +43,18 @@ public byte[] GetLocalMachineLocation(AbsolutePath cacheRoot)
 
             string networkPathRoot = cacheRoot.Path.Replace(":", "$");
 
-            return
-                Encoding.UTF8.GetBytes(
-                    Path.Combine(@"\\" + _localMachineName, networkPathRoot).ToUpperInvariant());
+            if (OperatingSystemHelper.IsWindowsOS)
+            {
+                // Only unify paths along casing if on Windows
+                networkPathRoot = Path.Combine(@"\\" + _localMachineName, networkPathRoot).ToUpperInvariant();
+            }
+            else
+            {
+                // Path.Combine ignores the first parameter if the second is a rooted path. To get the machine name before the rooted network path, the combination must be done manually.
+                networkPathRoot = @"\\" + _localMachineName + (networkPathRoot.StartsWith(Path.DirectorySeparatorChar.ToString()) ? string.Empty : Path.DirectorySeparatorChar.ToString()) + networkPathRoot;

```

Comment:
>DirectorySeparatorChar [](start = 152, length = 22)

on non-windows this will be `/`
so we'll end up with paths like `\\MACHINE/share\folder`
is this expected and okay?

Reply:
Good point. I'll change this from`\\MACHINE/share/folder` to `/MACHINE/share/folder`.

---
In reply to: [276399282](https://github.com/Microsoft/BuildXL/pull/156#discussion_r276399282) [](ancestors = 276399282)